### DISCUSSION TITLE
Surface normals and area

### DISCUSSION THREAD

**Discussion by jannav984** [su2code/SU2#2015] [Q&A]
Hello,

I'm working on FSI problem - coupling SU2 with Nastran solver. In this regards, I need to obtain surface normal and area around grid node to be able to calculate aerodynamic force. So, I have a question: 

Is there any way how to get: 
- either surface normal and area at all surface nodes,
- or aerodynamic force at all surface nodes?

Could it be done using python wrapper?

Thanks for answers.

Sincerely,

Jan

**Comment by Nicola-Fonzi** [comment]
Dear @jannav984 , I have coupled SU2 with Nastran using the python wrapper. Have a look at the tutorial in https://su2code.github.io/tutorials/Dynamic_FSI_Python/
Let me know if it helps

**Reply by jannav984** [reply]
Dear @Nicola-Fonzi,
thank you. I went through your tutorial. The problem is, that I want to solve nonlinear static deformation (large deformations) in Nastran. The modal approach is linear. 
Nevertheless, your coupling have some common solution with my idea - RBF for interface definition. Maybe, I can get some inspiration how to implement my stuff to SU2 from the code you developed.

Jan

**Comment by Nicola-Fonzi** [comment]
Dear Jan,

Yes, we use a linear approach. We have been considering the extension for some time, but had no time yet. We have an extension to nonlinear problems but it is for abaqus and not for Nastran. However, I believe that the procedure is very similar. If you want we can get in contact and we can discuss about what you need. Maybe I have some code to share that is of help. The extension for nonlinear problems is not yet pushed.

Let me know

Best

Nicola Fonzi

**Reply by jannav984** [reply]
Dear Nicola,

it would be great if we can discuss this topic. I write you an email.

Thank you.

Regards,
Jan
